spark/src/main/java/org/apache/iceberg/spark/ExceptionUtil.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark;
+
+import java.io.IOException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.RuntimeIOException;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.spark.sql.AnalysisException;
+
+public class ExceptionUtil {
```

Comment:
Originally, I thought about a private method in `SparkTableUtil` that would accept a lambda but having a separate class probably makes more sense as we can reuse this work in the actions in the future. I am not sure about the name, though. Since we are handling only Spark exceptions, maybe it makes sense to call it `SparkExceptionUtil`. I'll be interested to hear other opinions on this.

Reply:
`SparkExceptionUtil` does make sense. I will wait for any consensus and rename it. Rest of the comments are addressed.

Reply:
Renamed `ExceptionUtil` to `SparkExceptionUtil`